onnx_chainer/context.py

```diff
@@ -1,13 +1,29 @@
+import chainer
+
 from onnx_chainer import onnx_helper
 
 
 class Context(object):
+    """context of converter
+
+    This context shares names during exporting.
+
+    Arguments:
+        name_list (dict): list of being exported as ONNX node name keyed by
+            instance ID. When the target variable is ``chainer.Variable`` or
+            ``chainer.Parameter``, instance ID of data array held by the
+            variable is also put as key, because some functions has data array
+            instead of variable.
+
+    """
 
     def __init__(self, model):
         self.name_list = dict()
         for name, param in model.namedparams():
+            assert isinstance(param, chainer.Parameter)
             onnx_name = onnx_helper.cleanse_param_name(name)
             self.name_list[str(id(param))] = onnx_name
+            self.name_list[str(id(param.array))] = onnx_name
```

Comment:
Maybe add a helper function like `set_array_name` and do something like

```
self.name_list[id(param)] = onnx_name
if isinstance(param, (chainer.Variable, chainer.Parameter)):
  self.name_list[id(param.array)] = onnx_name
```

?

Also, it'd be better to mention why we need ids of both `Variable` and `ndarray`. Perhaps we can say something like "some functions like F.where internally unwraps Variable"?

Reply:
I'm also wondering if it's possible to use `id(a)` instead of `str(id(a))`.

Reply:
I fixed to aggregate `set_name` and rename `str(id(a))` to `id(a)`.

And I modified `Context` description to more clarify object of putting `ndarray` instance id. https://github.com/chainer/onnx-chainer/blob/d68fad0f7c09ad540e916b57d29a64595f2201ff/onnx_chainer/context.py#L12-L16